src/EnergyPlus/OutsideEnergySources.cc

```diff
@@ -331,23 +367,55 @@ void OutsideEnergySourceSpecs::size(EnergyPlusData &state)
     std::string typeName;
     if (this->EnergyType == DataPlant::PlantEquipmentType::PurchChilledWater) {
         typeName = "Cooling";
-    } else { // Heating
+    } else if (this->EnergyType == DataPlant::PlantEquipmentType::PurchHotWater) {
         typeName = "Heating";
+    } else { // DataPlant::PlantEquipmentType::PurchSteam
+        typeName = "HeatingSteam";
     }
 
```

Comment:
Can make an `auto &loop = state.dataPlnt->PlantLoop(this->plantLoc.loopNum);` shortcut and use it throughout here.